src/main/java/org/ngafid/flights/process/CSVFileProcessor.java

```diff
@@ -46,105 +54,441 @@ private CSVFileProcessor(Connection connection, byte[] bytes, String filename, P
+                // Populate columns with data from the flightRows
+                for (String[] row : flightRows) {
+                    if (row.length < firstRow.length) {
+                        break;
+                    }
+                    for (int j = 0; j < row.length; j++) {
+                        columns.get(j).add(row[j]);
+                    }
+                }
+                // G5 doesn't have local date time and utcOffset column, we calculate them here
+                TimeUtils.LocalDateTimeResult localDateTimeResult = null;
+                if(isG5FlightRecorder) {
+                    Integer utcDateIndex = g5headerIndices.get("UTC Date");
+                    Integer utcTimeIndex = g5headerIndices.get("UTC Time");
+                    Integer latitudeIndex = g5headerIndices.get("Latitude");
+                    Integer longitudeIndex = g5headerIndices.get("Longitude");
+
+                    StringTimeSeries utcDateSeries = new StringTimeSeries(headers.get(utcDateIndex), dataTypes.get(utcDateIndex), columns.get(utcDateIndex));
+                    StringTimeSeries utcTimeSeries = new StringTimeSeries(headers.get(utcTimeIndex), dataTypes.get(utcTimeIndex), columns.get(utcTimeIndex));
+                    DoubleTimeSeries latitudeTimeSeries = new DoubleTimeSeries(headers.get(latitudeIndex), dataTypes.get(latitudeIndex), columns.get(latitudeIndex));
+                    DoubleTimeSeries longitudeTimeSeries = new DoubleTimeSeries(headers.get(longitudeIndex), dataTypes.get(longitudeIndex), columns.get(longitudeIndex));
+                    localDateTimeResult = TimeUtils.calculateLocalDateTimeFromTimeSeries(timeZoneMap, utcDateSeries,utcTimeSeries,latitudeTimeSeries,longitudeTimeSeries);
+
+                    stringTimeSeries.put("Lcl Date", new StringTimeSeries("Lcl Date", "yyyy-MM-dd", localDateTimeResult.getLocalDates()));
+                    stringTimeSeries.put("Lcl Time", new StringTimeSeries("Lcl Time", "HH:mm:ss", localDateTimeResult.getLocalTimes()));
+                    stringTimeSeries.put("UTCOfst", new StringTimeSeries("UTCOfst", "hh:mm", localDateTimeResult.getUtcOffsets()));
+                }
+
+                // Populate doubleTimeSeries and stringTimeSeries
+                int startIndex = isG5FlightRecorder ? 2 : 0; // Skip first two columns (UTC date/time) if it is G5 flight recorder
+                for (int j = startIndex; j < columns.size(); j++) {
+                    var column = columns.get(j);
+                    var name = headers.get(j);
+                    var dataType = dataTypes.get(j);
+
+                    try {
+                        Double.parseDouble(column.get(0));
+                        doubleTimeSeries.put(name, new DoubleTimeSeries(name, dataType, column));
+                    } catch (NumberFormatException e) {
+                        stringTimeSeries.put(name, new StringTimeSeries(name, dataType, column));
+                    }
+                }
+
```

Comment:
So this should happen before anything is split. I think it makes the most sense to:

1. Parse all of the data, then
2. Crawl through the dates, looking for places where we may want to split.
3. Do the split.

The main parse should be in this parse method, but the splitting stuff should be in its own method. I think it will look something like this:

```
Stream<FlightBuilder> split(Map<String, DoubleTimeSeries> doubleTimeSeries, Map<String, StringTimeSeries> stringTimeSeries, ArrayList<String[]> data);
```

You may also need to pass some additional data w/ instance variables.